### PR TITLE
#2521 - Concrete projection a hyperrectangular cartesian product

### LINKED ISSUE #2521: Concrete projection a hyperrectangular cartesian product
```julia
X = Interval(1, 2) × Hyperrectangle(zeros(2), ones(2))

LazySets.project(X, [1, 3])

AssertionError: package 'Polyhedra' not loaded (it is required for executing `linear_map with elimination`)
```
but the result is as hyperrectangle.

### PR DESCRIPTION
**Pull request by SebastianGuadalupe**
Closes #2521.

### CODE REVIEW

File: test/unit_CartesianProduct.jl [hunk 1]
```diff
@@ -383,6 +383,10 @@ for N in [Float64, Float32]
     @test_throws AssertionError is_intersection_empty(cpa1, Universe{N}(3))
     @test_throws AssertionError is_intersection_empty(Universe{N}(5), cpa2)
 
+    # projection
+    cp = LazySets.Interval(N(0), N(2)) × Hyperrectangle(N[2, 3], N[1, 1])
+    @test _project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
+    @test _project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
```
Comment (schillic) [inline test/unit_CartesianProduct.jl:389]:
```suggestion
    @test project(cp, 1:2) == Hyperrectangle(N[1, 2], N[1, 1])
    @test project(cp, 2:3) == Hyperrectangle(N[2, 3], N[1, 1])
```

File: src/LazyOperations/CartesianProduct.jl [hunk 2]
```diff
@@ -294,3 +294,34 @@ end
 function concretize(cp::CartesianProduct)
     return cartesian_product(concretize(cp.X), concretize(cp.Y))
 end
+
+"""
+    project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
```
Comment (schillic) [inline src/LazyOperations/CartesianProduct.jl:299]: `vars` is called `block` in the documentation. I suggest to use the same name that is used in other methods.

File: src/LazyOperations/CartesianProduct.jl [hunk 3]
```diff
@@ -294,3 +294,34 @@ end
...
+
+Concrete projection of a cartesian product between an interval and an
```
Comment (schillic) [inline src/LazyOperations/CartesianProduct.jl:301]:
```suggestion
Concrete projection of a Cartesian product between an interval and a
```

File: src/LazyOperations/CartesianProduct.jl [hunk 4]
```diff
@@ -294,3 +294,34 @@ end
...
+hyperrectangle.
+
+### Input
+
+- `cp`       -- cartesian product between an interval and an hyperrectangle
```
Comment (schillic) [inline src/LazyOperations/CartesianProduct.jl:306]:
```suggestion
- `cp`       -- Cartesian product between an interval and a hyperrectangle
```

File: src/LazyOperations/CartesianProduct.jl [hunk 5]
```diff
@@ -294,3 +294,34 @@ end
...
+- `block`    -- block structure, a vector with the dimensions of interest
+
+### Output
+
+A set representing the projection of the cartesian product `cp` on the
+dimensions specified by `block`.
+"""
+function project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
+    Δt = cp.X
```
Comment (schillic) [inline src/LazyOperations/CartesianProduct.jl:315]: Please use a different name for `Δt`.

**Review by schillic — APPROVED** [review]

**Review by mforets — APPROVED** [review]
lgtm

**Pull request merged**